### PR TITLE
[ONNX] Update the script for version updates

### LINKED ISSUE #81075: [ONNX] Support opset 17 operators
Tracking issue on the onnx opset 17 support. Contributions welcome.

- [x] Add version 17 as a known version #83287 
- [ ] fft
  - [ ] ifft
  - [ ] fft_fft
  - [ ] fft_ifft
  - [ ] fft_rfft
  - [ ] fft_irfft
  - [ ] fft_hfft
  - [ ] fft_ihfft
  - [ ] fft_fft2
  - [ ] fft_ifft2
  - [ ] fft_rfft2
  - [ ] fft_irfft2
  - [ ] fft_hfft2
  - [ ] fft_ihfft2
  - [ ] fft_fftn
  - [ ] fft_ifftn
  - [ ] fft_rfftn
  - [ ] fft_irfftn
  - [ ] fft_hfftn
  - [ ] fft_ihfftn
  - [ ] fft_fftfreq
  - [ ] fft_rfftfreq
  - [ ] fft_fftshift
  - [ ] fft_ifftshift
- [x] stft #83944 
- [ ] istft
- [ ] bartlett_window
- [ ] blackman_window
- [ ] hamming_window
- [ ] hann_window
- [ ] kaiser_window
- [x] LayerNormalization #84293

Closes #80834

### PR DESCRIPTION
**Pull request by justinchuby**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #83283

This PR updates the `tools/onnx/update_default_opset_version.py` script to ensure files are edited correctly to prepare for the opset 17 support in torch.onnx.

- (clean up) Move script to `main()`
- Add an `--skip_build` option to avoid building pytorch if we want to rerun the process due to errors after compilation is done
- Update to edit the correct files now that the onnx files were refactored

### CODE REVIEW

**Review by thiagocrepaldi — COMMENTED** [review]
Hi @justinchuby, please provide a clear and concise description of the bug being handled or the feature/refactored being proposed. The PR description gives context for a faster review. 

If this is a bug fix, describe the bug, provide a minimal repro, describe the strategy to fix and provide unit tests.
If this is a feature/redesign (seems to be the case), outline the motivation for the proposal. Is your code related to a specific problem? e.g., "I'm working on X and would like Y to be possible". If this is related to another GitHub issue, please link here too.

**Comment by justinchuby** [comment]
Done. PTAL

File: tools/onnx/update_default_opset_version.py [hunk 1]
```diff
@@ -77,18 +29,80 @@ def read_sub_write(path: str, prefix_pat: str) -> None:
...
+    print("Updating operator .expect files")
+    subprocess.check_call(
+        ("python", os.path.join("test", "onnx", "test_operators.py"), "--accept"),
+    )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--skip_recompile", action="store_true", help="recompile pytorch"
```
Comment (thiagocrepaldi) [inline tools/onnx/update_default_opset_version.py:106]:
```suggestion
        "--skip_build", action="store_true", help="Skips pytorch (re-)build"
```
`recompile` is an unusual term whereas setuptools used by pytorch has targets such as `build`, `install`, `develop`, `bdist` and `clean`. Maybe we should use `build` instead

Also the flag was `--skip_recompile` but the help `recompile pytorch`, one is the opposite of the other :)

**Review by thiagocrepaldi — CHANGES_REQUESTED** [review]

File: tools/onnx/update_default_opset_version.py (hunk 1 above)
Comment (justinchuby) [reply tools/onnx/update_default_opset_version.py:106]: Thanks for catching!

File: tools/onnx/update_default_opset_version.py [hunk 2]
```diff
@@ -77,18 +29,84 @@ def read_sub_write(path: str, prefix_pat: str) -> None:
...
+    print("Updating operator .expect files")
+    subprocess.check_call(
+        ("python", os.path.join("test", "onnx", "test_operators.py"), "--accept"),
+    )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--skip_build", action="store_true", help="Skip building pytorch"
```
Comment (AllenTiTaiWang) [inline tools/onnx/update_default_opset_version.py:110]: Please remember to update the change to your description as well.

**Review by titaiwangms — APPROVED** [review]
🍺

File: tools/onnx/update_default_opset_version.py (hunk 2 above)
Comment (justinchuby) [reply tools/onnx/update_default_opset_version.py:110]: Done

File: tools/onnx/update_default_opset_version.py (hunk 1 above)
Comment (justinchuby) [reply tools/onnx/update_default_opset_version.py]: PTAL

**Review by thiagocrepaldi — APPROVED** [review]

**Comment by justinchuby** [comment]
cc @abock for approval

**Review by abock — APPROVED** [review]

**Pull request closed without merging**